woocommerce/google-listings-and-ads
# Add CombinedShipping component and apply it to the edit campaign page.

### Changes proposed in this Pull Request:


Implements https://github.com/woocommerce/google-listings-and-ads/issues/335 for edit flow.

### Screenshots:

![image](https://user-images.githubusercontent.com/17435/111087688-443b0400-8523-11eb-8fcd-c389b2a03832.png)


### Detailed test instructions:

1.  `git co && npm run build`
2.  Visit [Edit free campaign page](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fedit-free-campaign&programId=123&pageStep=2)
3.  Check if the look and shipping form behavior matches requirements from #335


## Still to be done:

~~Fix the API integration in regards to property used to store the combined shipping value.
Currently, it uses the legacy `shipping_rate` for it https://github.com/woocommerce/google-listings-and-ads/pull/343/files#diff-1e388a01205ba313fbab37402149798c4c1952a9b80ee909c05ced64c01fe196R58~~

### Doubts

1. ~~Confirm the gap width - right now, it has the smaller gap as proposed here https://github.com/woocommerce/google-listings-and-ads/issues/335#issuecomment-797641718~~
    - Eason edited: Jill had confirmed the gap here https://github.com/woocommerce/google-listings-and-ads/issues/335#issuecomment-799135060
2. ~~I used `combined-shipping.js` and `combined-shipping.scss` to make my development easier, we could change both to `index` to match the existing convention.~~
    - Eason edited: moved and renamed
3. This PR has a mix of save-less shipping rate and auto-saved shipping time - that could be addressed in a separate PR.
